lib/excerpt_parser.rb

```diff
@@ -220,11 +219,7 @@ def characters(string, truncate: true, count_it: true, encode: true, before_stri
     @current_length += string.length if count_it
   end
 
-  def truncated_emoji_code?(string)
-    @strip_truncated_emoji_code && emoji?(string)
-  end
-
   def emoji?(string)
-    string.match?(/:\w+:/)
+    string.match?(/^:\w+:$/)
```

Comment:
I don't know if it matters in this particular case but in ruby it's preferred to use `\A` and `\Z` instead of `^` and `$`:

```rb
":zzz:".match?(/^:\w+:$/)
=> true
"123\n:zzz:\n456".match?(/^:\w+:$/)
=> true

":zzz:".match?(/\A:\w+:\Z/)
=> true
"123\n:zzz:\n456".match?(/\A:\w+:\Z/)
=> false
```

Reply:
It should not matter as we split the sentence into words and words are compared with regex but better safe than sorry, thank you :)